tutorials/persistance-comparison-series/File/file.md

```diff
@@ -0,0 +1,345 @@
+
+First, update the hit counts so that we can save three of them:
+
+```cs
+[SerializeField] private int hitCountUnmodified = 0;
+[SerializeField] private int hitCountShift = 0;
+[SerializeField] private int hitCountControl = 0;
+```
+
+We also want to use a different file name so we can look at both version next to each other:
+
+```cs
+private const string HitCountFileUnmodified = "hitCountFileExtended.txt";
+```
+
+The last field we need to define is the key that is pressed:
+
+```cs
+private KeyCode keyPressed = default;
+```
+
+The first thing we need to do is check if a key was pressed and which key it was. Unity offers an easy way to achieve this using the [`Input`](https://docs.unity3d.com/ScriptReference/Input.html) class's `GetKey` function. It checks if the given key was pressed or not. You can pass in the string for the key or to be a bit more safe, just use the `KeyCode` enum. We cannot use this in the `OnMouseClick()` when detecting the mouse click though:
+
+> Note: Input flags are not reset until Update. You should make all the Input calls in the Update Loop.
+
+Add a new method called `Update()` (1) which is called in every frame. Here we need to check if the `Shift` or `Control` key was pressed (2) and if so, save the corresponding key in `keyPressed` (3). In case none of those keys was pressed (4) we consider it unmodified and reset `keyPressed` to its `default` (5).
+
+```cs
+private void Update() // 1
+{
+    // Check if a key was pressed.
+    if (Input.GetKey(KeyCode.LeftShift)) // 2
+    {
+        // Set the LeftShift key.
+        keyPressed = KeyCode.LeftShift; // 3
+    }
+    else if (Input.GetKey(KeyCode.LeftControl)) // 2
+    {
+        // Set the LeftControl key.
+        keyPressed = KeyCode.LeftControl; // 3
+    }
+    else // 4
+    {
+        // In any other case reset to default and consider it unmodified.
+        keyPressed = default; // 5
+    }
+}
+```
+
+Now to saving the data when a click happens:
+
+```cs
+private void OnMouseDown() // 6
+{
+    // Check if a key was pressed.
+    if (keyPressed == KeyCode.LeftShift) // 7
+    {
+        // Increment the Shift hit count.
+        hitCountShift++; // 8
+    }
+    else if (keyPressed == KeyCode.LeftControl) // 7
+    {
+        // Increment the Control hit count.
+        hitCountControl++; // 8
+    }
+    else // 9
+    {
+        // If neither Shift nor Control was held, we increment the unmodified hit count.
+        hitCountUnmodified++; // 10
+    }
+
+    // 11
+    // Create a string array with the three hit counts.
+    string[] stringArray = {
+        hitCountUnmodified.ToString(),
+        hitCountShift.ToString(),
+        hitCountControl.ToString()
+    };
+
+    // 12
+    // Save the entries, line by line.
+    File.WriteAllLines(HitCountFileUnmodified, stringArray);
```

Comment:
I don't think we're showing an example of how to read the extended file format.

Reply:
Indeed. I turned the structure around here since I thought it might be nicer to not follow the game flow but first save and then read (we only ever start the game after everything is done anyway) and then I must have forgotten to read it after saving it. :D

9a5844e